### PR TITLE
capi_dec: avoid two SharedPtr managing one pointer

### PR DESCRIPTION
**Pull request by wudping**
If we use "frame.reset(tmp, freeFrame)" to set "tmp" to new SharedPtr
variable "frame". That pointer is managed by SharedPtrHold::frame. It
will lead to double-free when their reference count is decreased to 0.

to fix vp9 clips decoding issues in --capi mode.

Signed-off-by: wudping <dongpingx.wu@intel.com>

### CODE REVIEW

**Comment by wudping** [comment]
please help to review, Thanks.

File: tests/vppinputdecodecapi.cpp [hunk 1]
```diff
@@ -67,7 +73,11 @@ bool VppInputDecodeCapi::read(SharedPtr<VideoFrame>& frame)
     while (1) {
         VideoFrame* tmp = decodeGetOutput(m_decoder);
         if (tmp) {
-            frame.reset(tmp, freeFrame);
+            SharedPtrHold* hold = (SharedPtrHold*)tmp->user_data;
```
Comment (xuguangxin) [inline tests/vppinputdecodecapi.cpp]: why we need this? we already hook the free to freeHold

**Comment by uartie** [comment]
please rebase

**Comment by wudping** [comment]
rebased, Thanks @uartie .

**Comment by uartie** [comment]
@wudping what's the status on this?

**Comment by wudping** [comment]
@uartie , I will close this pull-request. And I will re-open it if necessary.

**Pull request closed without merging**